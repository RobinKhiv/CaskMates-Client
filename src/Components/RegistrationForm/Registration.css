.registration-form{
    text-align: center;
    border-radius: 3px; 
    background-color: rgba(233, 219, 204, 0.788);
    padding-bottom: 10px;
    border: 1px solid #7220021f;
}
.landing-registration-page{
    color: #260000;
}
.register-title{
    text-align: center;
    margin-top: 5px;
    margin-bottom: 10px;
}
.registration-container{
   margin-right: auto;
}
.user_name, .full_name, .nickname, .password{
    text-align: center;
    margin-bottom: .3em;
}
.branding-quote{
    width: 75%;
    margin-left: auto;
    margin-right: auto;
}
@media only screen and (max-width: 640px) {
    .registration-container{
      margin-left: auto;
      padding-left: 1em;
      padding-right: 1em;
      margin-bottom: 4px;
    }
}