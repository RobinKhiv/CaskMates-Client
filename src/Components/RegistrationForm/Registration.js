import React, { Component } from 'react'
import AuthApiService from '../../Services/auth-api-service';

export default class Registration extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  }
  state = { error: null }

  handleRegistrationSubmit = (ev) => {
    ev.preventDefault();
    const {user_name, full_name, nickname, password} = ev.target;
    this.setState({ error: null })
    AuthApiService.postUser({
        user_name: user_name.value,
        full_name: full_name.value,
        nickname: nickname.value,
        password: password.value
    })
    .then(user => {
    full_name.value = ''
    nickname.value = ''
    user_name.value = ''
    password.value = ''
    this.props.onRegistrationSuccess()
    })
    .catch(res => {
        this.setState({error: res.error})
    }) 
  }

  render() {
    const {error} = this.state
    return (
      <form className="registration-form-container row" onSubmit={this.handleRegistrationSubmit}>
        <div role="alert">
          {error && <p className="red">{error}</p>}
        </div>
        <h2>Register</h2>
        <label htmlFor="user_name">Username : </label>
        <input type="text" name="user_name" className="user_name" required/><br/>
        <label htmlFor="full_name">Full Name : </label>
        <input type="text" name="full_name" className="full_name" required/><br/>
        <label htmlFor="nickname">Nickname : </label>
        <input type="text" name="nickname" className="nickname"/><br/>
        <label htmlFor="password">password : </label>
        <input type="text" name="password" className="password" required/><br/>
        <input type="submit" name="submit" value="Register"/>
      </form>
    )
  }
}
