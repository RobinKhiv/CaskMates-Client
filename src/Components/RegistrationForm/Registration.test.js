import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Registration from './Registration';

describe('<Registration />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render( <Registration /> , div);
        ReactDOM.unmountComponentAtNode(div);
    });

  it('renders without crashing', () => {
    const tree = renderer
      .create(<Registration />)
      .toJSON();
    expect(tree).toMatchSnapshot(); 
  });
});