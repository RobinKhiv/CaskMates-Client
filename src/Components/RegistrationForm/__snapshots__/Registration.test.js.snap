// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Registration /> renders without crashing 1`] = `
<form
  className="registration-form col-3"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <h2>
    Register
  </h2>
  <label
    htmlFor="user_name"
  >
    Username : 
  </label>
  <input
    className="user_name"
    name="user_name"
    required={true}
    type="text"
  />
  <br />
  <label
    htmlFor="full_name"
  >
    Full Name : 
  </label>
  <input
    className="full_name"
    name="full_name"
    required={true}
    type="text"
  />
  <br />
  <label
    htmlFor="nickname"
  >
    Nickname : 
  </label>
  <input
    className="nickname"
    name="nickname"
    type="text"
  />
  <br />
  <label
    htmlFor="password"
  >
    password : 
  </label>
  <input
    className="password"
    name="password"
    required={true}
    type="text"
  />
  <br />
  <input
    name="submit"
    type="submit"
    value="Register"
  />
</form>
`;
